### PR TITLE
document `--no-problems-report`

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes https://github.com/gradle/gradle/issues/31266

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 1]
```diff
@@ -608,6 +608,12 @@ Set to `rich` to enable color and other rich output in the console output, regar
 +
 Set to `verbose` to enable color and other rich output like `rich` with output task names and outcomes at the lifecycle log level, (as is done by default in Gradle 3.5 and earlier).
 
+[[sec:command_line_problems]]
+=== Reporting problems
+
+`--no-problems-report`::
+Disable the generation of `build/reports/problems-report.html`
```
Comment (donat) [inline platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc:615]: - Can we add a quick description of what is the default behavior and link the user guide? Something like "Gradle generates an html summary containing the problems reported via the problems API [link]"

File: platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 2]
```diff
@@ -608,6 +608,12 @@ Set to `rich` to enable color and other rich output in the console output, regar
 +
 Set to `verbose` to enable color and other rich output like `rich` with output task names and outcomes at the lifecycle log level, (as is done by default in Gradle 3.5 and earlier).
 
+[[sec:command_line_problems]]
+=== Reporting problems
+
+`--no-problems-report`::
```
Comment (donat) [inline platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc:614]: Can we do `--problems-report` and `--problems-report false`?

Comment (donat) [reply platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc:614]: We should also document the default value like above.

File: platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 3]
```diff
@@ -608,6 +608,12 @@ Set to `rich` to enable color and other rich output in the console output, regar
 +
 Set to `verbose` to enable color and other rich output like `rich` with output task names and outcomes at the lifecycle log level, (as is done by default in Gradle 3.5 and earlier).
 
+[[sec:command_line_problems]]
+=== Reporting problems
```
Comment (donat) [inline platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc:612]: I'm not sure, but please consider adding similar content to ht `build environment` section. Some builds want to disable problem generation for sync. In that case, only a gradle property can work.

**Review by donat — CHANGES_REQUESTED** [review]

File: platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc (hunk 2 above)
Comment (reinsch82) [reply platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc:614]: only `--no-problems-report` and `--problems-report` 
similar to `--parallel` and `--no-parallel`

**Comment by donat** [comment]
@reinsch82 I still think users trying to configure problem report generation in the context of sync should find the corresponding Gradle property in the docs.

**Pull request merged**